arch/arm/core/aarch32/cortex_m/mpu/arm_core_mpu.c

```diff
@@ -82,6 +87,14 @@ static const struct z_arm_mpu_partition static_regions[] = {
 		.attr = K_MEM_PARTITION_P_RX_U_RX,
 		},
 #endif /* CONFIG_ARCH_HAS_RAMFUNC_SUPPORT */
+#if defined(CONFIG_CODE_DATA_RELOCATION)
```

Comment:
This is no different from the RAMFUNC support, at least in the sense of memory attributes. Would it be possible to group these things together?

Reply:
I don't know of a clean way to do this.

Reply:
This would make sense if the SRAM region was always the relocation destination, but it's not. ITCM is another possibility for some platforms.

Reply:
understood, thanks

Reply:
To address this issue I have created a new Kconfig called `CODE_DATA_RELOCATION_SRAM`. The changes made to `arm_code_mpu.c` file are now using` #if defined(CONFIG_CODE_DATA_RELOCATION_SRAM)` which allows the MPU entry to be added only when code is relocated to SRAM.